AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -35,99 +33,159 @@ protocol StatementModelConvertible {
-        for (index, column) in columns.enumerated()
-            where propertyPrefix == nil || column.starts(with: propertyPrefix!) {
-
-            let name: String = propertyPrefix != nil
-                ? column.replacingOccurrences(of: propertyPrefix!, with: "")
-                : column
-            if name.firstIndex(of: ".") != nil {
-                let propertyName = String(name.split(separator: ".").first!)
-                guard let associatedField = schema.field(withName: propertyName) else {
-                    preconditionFailure("""
-                    Field `\(propertyName)` not found on `\(schema.name)`.
-                    The property was found on the result set but was not defined in the
-                    `\(schema.name)` schema.
-                    """)
-                }
-                let associatedModelType = associatedField.requiredAssociatedModel
-                let associatedModel = try mapEach(row: row,
-                                                  to: associatedModelType,
-                                                  cache: &cache,
-                                                  fieldName: propertyName)
-                values[propertyName] = associatedModel
-            } else if let field = schema.field(withName: name) {
-                values[name] = try SQLiteModelValueConverter.convertToSource(from: row[index],
-                                                                             fieldType: field.type)
-            } else {
+    func convert(row: Element,
+                 to modelType: Model.Type,
+                 using statement: SelectStatement) throws -> ModelValues {
+        let columnMapping = statement.metadata.columnMapping
+        let modelDictionary = ([:] as ModelValues).mutableCopy()
+        for (index, value) in row.enumerated() {
+            let column = columnNames[index]
+            guard let (schema, field) = columnMapping[column] else {
                 logger.debug("""
-                A column named \(name) was found in the result set but no field on
-                \(schema.name) could be found with that name and it will be ignored.
+                A column named \(column) was found in the result set but no field on
+                \(modelType.modelName) could be found with that name and it will be ignored.
                 """)
+                continue
+            }
+
+            let modelValue = try SQLiteModelValueConverter.convertToSource(
+                from: value,
+                fieldType: field.type
+            )
+            modelDictionary.updateValue(modelValue, forKeyPath: column)
+
+            // create lazy list for "many" relationships
+            if let id = modelValue as? String, field.isPrimaryKey {
+                let associations = schema.fields.values.filter {
+                    $0.isArray && $0.hasAssociation
+                }
+                let prefix = field.name.replacingOccurrences(of: field.name, with: "")
```

Comment:
what is the prefix here? isn't it just ""?

Reply:
it depends on how many levels deep you are in the object graph. It starts with `""`, it becomes `"post."` then `"post.blog."`